tensordict/tensorclass.py

```diff
@@ -191,25 +195,41 @@ def __torch_function__(
     cls.__ne__ = _ne
     cls.__or__ = _or
     cls.__xor__ = _xor
-    cls.set = _set
-    cls.set_at_ = _set_at_
-    cls.del_ = _del_
-    cls.get = _get
-    cls.get_at = _get_at
-    cls.unbind = _unbind
-    cls.state_dict = _state_dict
-    cls.load_state_dict = _load_state_dict
-    cls._memmap_ = _memmap_
+    if not hasattr(cls, "set"):
+        cls.set = _set
+    if not hasattr(cls, "set_at_"):
+        cls.set_at_ = _set_at_
+    if not hasattr(cls, "del_"):
+        cls.del_ = _del_
+    if not hasattr(cls, "get"):
+        cls.get = _get
+    if not hasattr(cls, "get_at"):
+        cls.get_at = _get_at
+    if not hasattr(cls, "unbind"):
+        cls.unbind = _unbind
+    if not hasattr(cls, "state_dict"):
+        cls.state_dict = _state_dict
+    if not hasattr(cls, "load_state_dict"):
+        cls.load_state_dict = _load_state_dict
+    if not hasattr(cls, "_memmap_"):
+        cls._memmap_ = _memmap_
+    if not hasattr(cls, "share_memory_"):
+        cls.share_memory_ = _share_memory_
```

Comment:
Ideally we'd like that if the class has this attribute and the built-in calls `super().foo(...)`, then this falls back on the implementation we're setting here. 
I'm not sure of how to make this happen...

Reply:
The best if we don't do #663 (the main blocker is the python version on my side) would be to discourage usage of `super().smth()` in `tensorclasses` in favour of something different, like
```
@tensorclass
class A:
   X: torch.Tensor
   
   def __getitem__(self, item):
       # your code here
       return super_method(self, "__getitem__")(item)
```
and `super_method` would be defined within `tensorclass.py` as a method falling back on the custom `tensorclass` implementation